src/language/branches.md

```diff
@@ -0,0 +1,229 @@
+}
+```
+For an empty case that does not fall through, use [`break`][break] for its body.
+
+### Switch expressions
+
+Switch expressions are similar to switch statements, but you can use them anywhere
+you can use an expression. They produce a value based on the expression body of
+whichever case matches.
+
+Switch expression syntax differs slightly from switch statements:
+
+- Cases _do not_ start with the `case` keyword.
+- A case body is a single expression instead of a series of statements.
+- Each case must have a body; there is no implicit fallthrough for empty cases.
+- Case patterns are separated from their bodies using `=>` instead of `:`.
+- Cases are separated by `,` (and an optional trailing `,` is allowed).
+- Default cases can _only_ use `_`, instead of allowing both `default` and `_`.
+
+Switch expressions allow you to rewrite a _statement_ like this:
+
+```dart
+// Where slash, star, comma, semicolon, etc., are constant variables...
+
+switch (charCode) {
+  case slash || star || plus || minus:    // Logical-or pattern
+    token = operator(charCode);
+  case comma || semicolon:                // Logical-or pattern
+    token = punctuation(charCode);
+  case >= digit0 && <= digit9:            // Relational and logical-and patterns
+    token = number();
+  default:
+    throw invalid();
+}
+```
+
+Into an _expression_, like this:
+
+```dart
+token = switch (charCode) {
+  slash || star || plus || minus => operator(charCode),
+  comma || semicolon => punctuation(charCode),
+  >= digit0 && <= digit9 => number(),
+  _ => throw invalid()
+};
+```
+
+### Exhaustiveness checking
+
+Exhaustiveness checking is a feature of switches that reports a compile-time
+error if it's possible for a value entering the switch to not match any of the cases.
+
+```dart
+bool? b = false;
+// Non-exhaustive switch on bool?, missing case to match null possiblity:
+switch (b) {
+  case true: print('yes');
+  case false: print('no');
+}
+```
+
+A default case (`default` or `_`) covers all possible values, making a switch on
+any type exhaustive.
+
+[Enums][enum] and [sealed types][sealed] are particularly useful for switches
+because, even without a default case, their possible values are known and fully
+enumerable. Use the [`sealed` modifier][sealed] on a class to enable
+exhaustiveness checking when switching over subtypes of that class:
+
+```dart
+sealed class Shape {
+  double calculateArea();
+}
+
+class Square extends Shape { ... }
+class Circle extends Shape { ... }
+
+// ...
+
+double calculateArea(Shape shape) =>
+  switch (shape) {
+    Square(length: var l) => l * l,
+    Circle(radius: var r) => math.pi * r * r
+  };
+```
+
+If a new subclass of `Shape` is ever added, exhaustiveness 
+checking will tell you about the missing subtype, because your switch is incomplete.
+This allows you to use Dart in a somewhat 
+[functional algebraic datatype style](https://en.wikipedia.org/wiki/Algebraic_data_type). 
```

Comment:
**issue:** Write in active voice per [GSG](https://developers.google.com/style/voice). 

```suggestion
If you add a new subclass of `Shape`,
exhaustiveness checking reports the missing subtype.
The check flags your `switch` as incomplete.
This allows you to use Dart in a somewhat 
[functional algebraic datatype style](https://en.wikipedia.org/wiki/Algebraic_data_type). 
```

Reply:
The benefit here is that if a subclass is added _unbeknownst to you_, exhaustiveness checking would alert you to it, so using "you" as the actor doesn't work for the selling point. I tried this:

> If anyone were to add a new subclass of `Shape`, this `switch` expression would 
be incomplete. Exhaustiveness checking would inform you of the missing subtype.